### DISCUSSION TITLE
I can't see the source code of the programme I want to debug?

### DISCUSSION THREAD

**Discussion by deetl** [eteran/edb-debugger#827] [Q&A] [answered]
Sorry for the stupid question, I'm sure I'm missing something completely trivial. 
I have a small C program compiled with gcc -g. When I load it in gdb, I see the source code and assembly code, all good. But when I open the same binary in edb, I get the assembly code but don't see the C code. But everything runs cleanly.

What am I missing? I'm sure I'm missing something completely trivial.

**Comment by eteran** [comment, marked as answer]
So, edb is specifically a reverse engineering debugger, it is not meant to be used as a "normal" debugger.

Because of this, it is operating on the assumption that you DON'T have the source code and are trying to understand someone else's program by looking at the disassembly.

**Comment by epasveer** [comment]
@deetl   If you want to see both source and assembly, you may want to try [Seer](https://github.com/epasveer/seer). It likely doesn't have all the disassembly features that edb provides, though.

@eteran  sorry for  suggesting another debugger on your page :^)

**Reply by deetl** [reply]
@epasveer thanks for the tip and I had a look on Seer before and it is really a very nice and comfortable debugger. But my problem is, I need a debugger for giving some lectures. So I need a very clean and simple interface (so no Ghidra), the possibility to execute assembler step by step to explain things, a visualization of the stack (explain function calls) and helpers (e.g. displaying data behind a memory address or inline translation of hex to ASCII). So edb is close to perfect, maybe having the source code (if available) as comment would be enough.

Thanks everybody!

**Reply by epasveer** [reply]
No worries. All good.

**Reply by 10110111** [reply]
@deetl you might find GDB's TUI sufficient for your purposes. Being console-based, it's not a nice "GUI", but still usable for displaying source and disassembly simultaneously (in separate frames), and can show pretty backtrace (via `backtrace` command) as well as examine memory contents (via e.g. `x/10w $sp` command to look at 10 most recently pushed dwords in the stack).

**Reply by deetl** [reply]
Thanks @10110111, gdb and gdbtui are sometimes my usual tools for debugging, i love command line tools. But when teaching, I need something that keeps the hurdle as small as possible. The students have to struggle with so many new concepts at the same time (assembler, registers, stack, compiler, etc.) so everything should be as obvious as possible. And it is much easier to follow a GUI where the stack is always visible and also breaks down the content somewhat than the gdb with its somewhat cryptic commands.
